### PR TITLE
AZP/RELEASE: Fix launch condition-v1.15.x

### PR DESCRIPTION
**Pull request by Alexey-Rivkin**
Cherry-picked from https://github.com/openucx/ucx/pull/9327.

### CODE REVIEW

File: buildlib/az-helpers.sh [hunk 1]
```diff
@@ -139,6 +140,16 @@ function az_module_unload() {
     module unload "${module}" || true
 }
 
+# Ensure that GPU is present
+check_gpu() {
```
[the remark this answers is not in the document]
Comment (Alexey-Rivkin) [reply buildlib/az-helpers.sh:144]: I stand corrected.
It's better to not add the GPU check in this PR.

**Review by yosefe — APPROVED** [review]
pls squash

**Review by yosefe — DISMISSED** [review]

**Review by yosefe — APPROVED** [review]
pls squash

**Comment by Alexey-Rivkin** [comment]
squashed

**Comment by Alexey-Rivkin** [comment]
io_demo tests fail, as this PR doesn't include the io_demo setters fix https://github.com/openucx/ucx/pull/9332

**Review by artemry-nv — APPROVED** [review]

**Review by yosefe — APPROVED** [review]

**Review by artemry-nv — APPROVED** [review]

**Pull request merged**